### PR TITLE
[DomCrawler] Add types to private properties

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/AbstractUriElement.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ abstract class AbstractUriElement
 
     /**
      * @param \DOMElement $node       A \DOMElement instance
-     * @param string      $currentUri The URI of the page where the link is embedded (or the base href)
+     * @param string|null $currentUri The URI of the page where the link is embedded (or the base href)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DomCrawler/AbstractUriElement.php]: for 4.4

**Comment by nicolas-grekas** [comment]
WDYT of psalm reports?

**Comment by stof** [comment]
For `getBaseHref`, it should indeed be documented as `@return string|null` (in 4.4). That's a phpdoc mistake.

for the `createSubcrawler` report, this would be solved by keeping the phpdoc `@var \DOMNode[]` on the `$nodes` property, which is more precise than the `array` type.
And for `getText`, I think it might be fixed by that phpdoc too (but not sure)

**Comment by derrabus** [comment]
> For `getBaseHref`, it should indeed be documented as `@return string|null` (in 4.4). That's a phpdoc mistake.

Same for `getUri()` btw. I'm going to backport those fixes.

> for the `createSubcrawler` report, this would be solved by keeping the phpdoc `@var \DOMNode[]` on the `$nodes` property, which is more precise than the `array` type.

Yes, I shouldn't have removed that.

> And for `getText`, I think it might be fixed by that phpdoc too (but not sure)

That error is gone now as well. It might have been caused by the type removal from `$nodes` as well. 🤷🏻‍♂️

**Comment by derrabus** [comment]
Backports have been merged up and this PR is ready again.

**Review by chalasr — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @derrabus.

**Pull request merged**